### PR TITLE
Pass through skipDuplicate

### LINKED ISSUE #1: Some Help Requested
Your documentation is thorough and excellent. I'm new to gh actions though so pardon me..

I'm writing a workflow to publish [rust-analyzer](https://github.com/rust-analyzer/rust-analyzer) to open-vsix.org. The repo is [here](https://github.com/hybras/Rust-Analyzer-Open-Vsix). I have a step to query github for the latest release and download the vsix file. It is successful. However, I'm having trouble passing it to the next step (yours, the publishing step). I'm getting `Error: Please specify either an extension file or a package path, but not both.` [I have not specified both keys](https://github.com/hybras/Rust-Analyzer-Open-Vsix/blob/master/.github/workflows/blank.yml#L34-L38). 

I understand you are under no obligation to help me, you published this action out of the goodness of your heart! Just close this issue then, I get it.

### LINKED ISSUE #38: Workflow fails when snippets package version is already in the registry
| publish-vscode-extension Version | Operating System |
| -------------------------------------- | ---------------- |
|  v1                                      |   ubuntu-latest               |

# Description
Adding more on [this issue](https://github.com/HaaLeo/publish-vscode-extension/issues/15).
IMHO the workflow should not error when a package version already exists in the registry.
I have a monorepo with multiple libraries and one snippets package.
Making changes to a library makes the workflow fail since the snippets package wasn't modified, and thus is still on the same version.

## Expected Behavior
The workflow should succeed also when the snippets extension is already in the registry.

## Actual Behavior
The workflow fails when changing other files in the workspace

image

# Steps to Reproduce
Everything is hosted [here](https://github.com/MintPlayer/mintplayer-ng-bootstrap)
Error is thrown [here](https://github.com/HaaLeo/publish-vscode-extension/blob/master/dist/index.js#L5098)
```
        if (extension && extension.versions) {
            const sameVersion = extension.versions.filter(v => v.version === manifest.version);
            if (sameVersion.length > 0) {
                if (options.skipDuplicate) {
                    util_2.log.done(`Version ${manifest.version} is already published. Skipping publish.`);
                    return;
                }
                if (sameVersion.some(v => v.targetPlatform === options.target)) {
/* Here */          throw new Error(`${description} already exists.`);
                }
            }
            try {
                await api.updateExtension(undefined, packageStream, manifest.publisher, manifest.name);
            }
            catch (err) {
```

This code comes [from here](https://github.com/microsoft/vscode-vsce/blob/main/src/publish.ts?fbclid=IwAR0aWrA4vG4CKGylXoE8BAE_qnKY7qkRvhLIrMiqktEWoZGXHjxEAlu1F8U#L185)

# Context
<details><pre><code>Run HaaLeo/publish-vscode-extension@v1
  with:
    packagePath: libs/mintplayer-ng-bootstrap-snippets
    pat: ***
    registryUrl: https://marketplace.visualstudio.com
    yarn: false
    dryRun: false
    noVerify: false
    preRelease: false
    dependencies: true
Specifically, running GitHub Action HaaLeo/publish-vscode-extension@v1.3.0.
Package the Extension
Publish the Extension
Error: Error: MintPlayer.mintplayer-ng-bootstrap-snippets v16.0.1 already exists.
    at _publish (/home/runner/work/_actions/HaaLeo/publish-vscode-extension/v1/dist/index.js:5098:[2](https://github.com/MintPlayer/mintplayer-ng-bootstrap/actions/runs/5065996309/jobs/9095228526#step:36:2)7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async publish (/home/runner/work/_actions/HaaLeo/publish-vscode-extension/v1/dist/index.js:50[3](https://github.com/MintPlayer/mintplayer-ng-bootstrap/actions/runs/5065996309/jobs/9095228526#step:36:3)[7](https://github.com/MintPlayer/mintplayer-ng-bootstrap/actions/runs/5065996309/jobs/9095228526#step:36:7):13)</code></pre></details>

### PR DESCRIPTION
**Pull request by PieterjanDeClippel**
# Summary

Pass through the `skipDuplicate` option to the `@vscode/vsce` methods.

# Changes

* Add `skipDuplicate` to the options
* Filter out `skipDuplicate` and pass to the `@vscode/vsce` `publishVSIX` method
* main: Read `skipDuplicate` from the Github Actions workflow step inputs

# Related Issues

Fixes #38 

# Motivation
My workspace contains multiple libraries and a snippets package.
Making modifications to the library, but not the snippets package, makes the workflow fail because the snippets package (which hasn't changed) is "already" in the marketplace registry

### CODE REVIEW

**Comment by PieterjanDeClippel** [comment]
@HaaLeo

File: ThirdPartyNotices.txt [hunk 1]
```diff
@@ -1,32 +1,32 @@
-THIRD-PARTY SOFTWARE NOTICES AND INFORMATION
```
Comment (HaaLeo) [inline ThirdPartyNotices.txt:1]: Can you pls undo the changes on this file? they shouldn't necessary

File: src/publish.ts [hunk 2]
```diff
@@ -17,14 +17,15 @@ async function publish(ovsxOptions: ActionOptions): Promise<void> {
 
 function _convertToVSCEPublishOptions(options: ActionOptions): VSCEPublishOptions {
     // Shallow copy of options
-    const { baseContentUrl, baseImagesUrl, pat, yarn: useYarn, noVerify, dependencies } = { ...options };
+    const { baseContentUrl, baseImagesUrl, pat, yarn: useYarn, noVerify, dependencies, skipDuplicate } = { ...options };
```
Comment (HaaLeo) [inline src/publish.ts:20]: Can you extend the tests pls to check for the `skipDuplicate` property?

**Review by HaaLeo — CHANGES_REQUESTED** [review]

**Comment by HaaLeo** [comment]
I had some spare time to add the feature myself. Nevertheless thank you a lot for your contribution. It feels horrible to close an unmerged PR

**Pull request closed without merging**